client/packages/system/src/Report/components/ReportArgumentsModal.tsx

```diff
@@ -32,12 +33,12 @@ export const ReportArgumentsModal: FC<ReportArgumentsModalProps> = ({
     monthsOverstock,
     monthsUnderstock,
     monthsItemsExpire,
+    ...JSON.parse((urlQuery?.['reportArgs'] ?? '{}') as string),
   });
   const [error, setError] = useState<string | false>(false);
 
   // clean up when modal is closed
   const cleanUp = () => {
-    setData({});
```

Comment:
Hopefully this doesn't cause any other problems. I've added the `key` prop to the Modal which should force a remount when the report type changes, which *should* work in lieu of clearing the data here.